app/views/recipes/_form.html.erb

```diff
@@ -48,6 +48,16 @@
     <%= form.check_box :is_favorite, rows: 4, class: "block shadow rounded-md border border-gray-200 outline-none px-2 py-2 mt-1" %>
   </div>
 
+    
+  <datalist id="categories">
+    <% Category.all.each do |c| %>
+        <option> <%= c.name %> </option>
+    <% end %>
+  </datalist>
+ 
+     <%= form.text_field ":categories", value: "", list: "categories", placeholder: "Enter categories, seperated by commas" %>  
```

Comment:
This was me trying different combinations, because when I submit I would get errors. One in the console or 'unpermitted parameter' even though :categories is in the recipe params. Or I would get instance of {} or [] error.